src/Module/Contact/Poke.php

```diff
@@ -0,0 +1,164 @@
+use Friendica\Core\System;
+use Friendica\Database\DBA;
+use Friendica\DI;
+use Friendica\Model;
+use Friendica\Network\HTTPException;
+use Friendica\Protocol\Activity;
+use Friendica\Util\XML;
+
+class Poke extends BaseModule
+{
+	public static function post(array $parameters = [])
+	{
+		if (!local_user() || empty($parameters['id'])) {
+			return self::postReturn(false);
+		}
+
+		$uid = local_user();
+
+		if (empty($_POST['verb'])) {
+			return self::postReturn(false);
+		}
+
+		$verb = $_POST['verb'];
+
+		$verbs = DI::l10n()->getPokeVerbs();
+		if (!array_key_exists($verb, $verbs)) {
+			return self::postReturn(false);
+		}
+
+		$activity = Activity::POKE . '#' . urlencode($verbs[$verb][0]);
+
+		$contact_id = intval($parameters['id']);
+		if (!$contact_id) {
+			return self::postReturn(false);
+		}
+
+		Logger::info('poke: verb ' . $verb . ' contact ' . $contact_id);
+
+		$contact = DBA::selectFirst('contact', ['id', 'name'], ['id' => $parameters['id'], 'uid' => local_user()]);
+		if (!DBA::isResult($contact)) {
+			return self::postReturn(false);
+		}
+
+		$a = DI::app();
+
+		$private = (!empty($_GET['private']) ? intval($_GET['private']) : Model\Item::PUBLIC);
+
+		$allow_cid     = ($private ? '<' . $contact['id']. '>' : $a->user['allow_cid']);
+		$allow_gid     = ($private ? '' : $a->user['allow_gid']);
+		$deny_cid      = ($private ? '' : $a->user['deny_cid']);
+		$deny_gid      = ($private ? '' : $a->user['deny_gid']);
+
+		$actor = $a->contact;
+
+		$uri = Model\Item::newURI($uid);
+
+		$arr = [];
+
+		$arr['guid']          = System::createUUID();
+		$arr['uid']           = $uid;
+		$arr['uri']           = $uri;
+		$arr['parent-uri']    = $uri;
+		$arr['wall']          = 1;
+		$arr['contact-id']    = $actor['id'];
+		$arr['owner-name']    = $actor['name'];
+		$arr['owner-link']    = $actor['url'];
+		$arr['owner-avatar']  = $actor['thumb'];
+		$arr['author-name']   = $actor['name'];
+		$arr['author-link']   = $actor['url'];
+		$arr['author-avatar'] = $actor['thumb'];
+		$arr['title']         = '';
+		$arr['allow_cid']     = $allow_cid;
+		$arr['allow_gid']     = $allow_gid;
+		$arr['deny_cid']      = $deny_cid;
+		$arr['deny_gid']      = $deny_gid;
+		$arr['visible']       = 1;
+		$arr['verb']          = $activity;
+		$arr['private']       = $private;
+		$arr['object-type']   = Activity\ObjectType::PERSON;
+
+		$arr['origin']        = 1;
+		$arr['body']          = '[url=' . $actor['url'] . ']' . $actor['name'] . '[/url]' . ' ' . $verbs[$verb][2] . ' ' . '[url=' . $contact['url'] . ']' . $contact['name'] . '[/url]';
+
+		$arr['object'] = '<object><type>' . Activity\ObjectType::PERSON . '</type><title>' . XML::escape($contact['name']) . '</title><id>' . XML::escape($contact['url']) . '</id>';
+		$arr['object'] .= '<link>' . XML::escape('<link rel="alternate" type="text/html" href="' . $contact['url'] . '" />') . "\n";
+
+		$arr['object'] .= XML::escape('<link rel="photo" type="image/jpeg" href="' . $contact['photo'] . '" />') . "\n";
```

Comment:
I guess you haven't checked, if this is still needed? Before accepting the PR, I want to have a look at the receiver as well. When we don't parse that object at all, we don't need it. (And in the past we got rid of most of this stuff. I have the feeling as if the only really important part is the `verb` field.

Reply:
No, I didn't touch this part at all. I was able to verify a poke item is being created and is private when it's supposed to.

Reply:
I'm sure that you hadn't touched this. But I'm always keen in optimizing stuff - and I like to remove unused stuff. Something that isn't there can't make problems.

Reply:
Update: It seems as if the object is indeed used. So we sadly can't remove it.